src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_NativeInteger.cs

```diff
@@ -825,6 +825,27 @@ static void Main()
             comp.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void EmitAttribute_PartialMethods()
+        {
+            var source =
+@"public partial class Program
+{
+    static partial void F1(System.IntPtr x);
+    static partial void F2(System.UIntPtr x) { }
+    static partial void F1(nint x) { }
+    static partial void F2(nuint x);
+}";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseDll.WithMetadataImportOptions(MetadataImportOptions.All), parseOptions: TestOptions.RegularPreview);
+            // Ideally should not emit any attributes. Compare with dynamic/object.
```

Comment:
>Ideally should not emit any attributes. [](start = 15, length = 39)

Why partial methods are special? Is this about private accessibility in general?

Reply:
The test is for merging of attributes across partial method parts. Private accessibility is tested elsewhere.

Reply:
> The test is for merging of attributes across partial method parts. 

Then why do we expect to emit no attributes?

---
In reply to: [400636238](https://github.com/dotnet/roslyn/pull/42745#discussion_r400636238) [](ancestors = 400636238)

Reply:
> Then why do we expect to emit no attributes?

For consistency with `dynamic` / `object` behavior.

Reply:
> For consistency with dynamic / object behavior.

Is there real intent behind that behavior as opposed to an omission/oversight? I would rather follow what we do for nullable and tuple names.

---
In reply to: [401056147](https://github.com/dotnet/roslyn/pull/42745#discussion_r401056147) [](ancestors = 401056147)